src/client/java/teammates/client/scripts/sql/DataMigrationForCourseEntitySql.java

```diff
@@ -530,4 +538,29 @@ private void migrateDeadlineExtension(DeadlineExtension oldDeadlineExtension,
         saveEntityDeferred(newDeadlineExtension);
     }
 
+    private void migrateUserAccounts(teammates.storage.sqlentity.Course newCourse, Map<String, User> userEmailToUserMap) {
+        List<Account> newAccounts = getAllAccounts(new ArrayList<String>(userEmailToUserMap.keySet()));
+        if (newAccounts.size() != userEmailToUserMap.size()) {
+            // #TODO Log error
+        }
+        for (Account account: newAccounts) {
+            User newUser = userEmailToUserMap.get(account.getEmail());
+            if (newUser == null) {
+                // #TODO Log error
+                continue;
+            }
+            newUser.setAccount(account);
```

Comment:
The method makes sense, but how are we persisting the users here?

Reply:
Good catch! I have move saveEntityDeffered for all user entities to this method!